pkg/operator/ceph/cluster/watcher.go

```diff
@@ -290,6 +327,38 @@ func listRBDPV(listPVs *corev1.PersistentVolumeList, cluster *cephv1.CephCluster
 	return listRbdPV
 }
 
+func listCephFSPV(listPVs *corev1.PersistentVolumeList, cluster *cephv1.CephCluster, cephFSVolumesInUse []string) []corev1.PersistentVolume {
+	var listCephFSPV []corev1.PersistentVolume
+
+	for _, pv := range listPVs.Items {
+		// Skip if pv is not provisioned by CSI
+		if pv.Spec.CSI == nil {
+			logger.Debugf("pv %q is not provisioned by CSI", pv.Name)
+			continue
+		}
+
+		if pv.Spec.CSI.Driver == fmt.Sprintf("%s.cephfs.csi.ceph.com", cluster.Namespace) {
+			// Ignore PVs that support multinode access (RWX, ROX), since they can be mounted on multiple nodes.
+			if pvSupportsMultiNodeAccess(pv.Spec.AccessModes) {
+				continue
+			}
+
+			if pv.Spec.CSI.VolumeAttributes["staticVolume"] == "true" || pv.Spec.CSI.VolumeAttributes["pool"] == "" {
+				logger.Debugf("skipping, static pv %q", pv.Name)
+				continue
+			}
+
+			for _, cephFSVolume := range cephFSVolumesInUse {
```

Comment:
We really need `cephFSVolumesInUse` to be a map to allow a quick lookup. If there are a lot of PVs and volumes in use, this is basically a double `for` loop that will cause a problem O(n^2).

Reply:
That's correct. Thanks, Travis
Updated the PR.
PTAL.